src/app/shared/image/image-upload.component.ts

```diff
@@ -98,6 +99,9 @@ export class ImageUploadComponent implements DoCheck {
         if (this.browserImage.width < this.minWidth || this.browserImage.height < this.minHeight) {
           this.imgError = `The image provided is only ${this.browserImage.width} x ${this.browserImage.height} px
                            but should be at least ${this.minWidth} x ${this.minHeight} px.`;
+        } else if (this.square && this.browserImage.width !== this.browserImage.height) {
+          this.imgError = `The image provided is ${this.browserImage.width} x ${this.browserImage.height} px
+                           but is required to be square and at least ${this.minWidth} x ${this.minHeight} px.`;
```

Comment:
Also, I wonder if this imgError should just be "The image provided must be square.", since they'll only hit this `if` clause if they make it past the first one. I think giving too much info/ numbers of px may be confusing.